inc/ThirdParty/Plugins/Optimization/AMP.php

```diff
@@ -115,4 +117,20 @@ public function disable_options_on_amp() {
 			remove_filter( 'wp_calculate_image_srcset', 'rocket_protocol_rewrite_srcset', PHP_INT_MAX );
 		}
 	}
+
+	/**
+	 * Rewrites URLs to the CDN URLs if allowed.
+	 * Rewrites URLs in srcset attributes if allowed.
+	 *
+	 * @since 3.5.5
+	 *
+	 * @param  string $html HTML content.
+	 * @return string       HTML content.
+	 */
+	public static function rewrite_cdn( $html ) {
+		$container      = apply_filters( 'rocket_container', '' );
+		$cdn_subscriber = $container->get( 'cdn_subscriber' );
```

Comment:
If the method is not `static`, then you could inject the CDN Subscriber. In doing so, you wouldn’t need to get the container and then a new instance of the `’cdn_subscriber’`. It would a little more performant and maintainable.

Reply:
How to inject the CDN Subscriber?

Reply:
@crystinutzaa You'll want to inject it into the constructor from the ServiceProvider that handles it.